### ISSUE TITLE
Kickstart UI: Verify LDAP Configuration for Test Group Name is not working

### ISSUE THREAD

**Issue by miclettej** [vmware-tanzu/tanzu-framework#1269]
What steps did you take and what happened:

I am trying to deploy TKG on vSphere with Pinniped and Dex providing LDAP integration. The LDAP Server is Microsoft Windows Active Directory running on Windows 2012 R2 DC. I wish to be able to use an LDAP 'group' to give multiple developers access to a cluster through a single ClusterRoleBinding.

Whilst populating the LDAP portion of the UI during the management cluster creation, I used the Verify LDAP Configuration option to validate my settings. I can get the Test User to pass each and every time, but I can never get the Test Group to pass. It consistently fails with "Unable to find group:". The group exists in my AD Directory Service, and has users associated it it, but the Verifier never seems to be able to find it.

What did you expect to happen:

I expect that I should be able to get the "Test Group Name" section of the Verifier to work as this would be an important part for ensuring LDAP is working before creating the cluster and giving developers access to clusters. I suspect that K8s platform admins do not want to be adding a ClusterRoleBinding for every individual developer, but instead give a Group access, and all users in the group get given the role binding. I cannot proceed / do not want to process with this part of my setup until I get the LDAP Verify section to work.

Anything else you would like to add:

Very difficult to troubleshoot this. There is no logging in the tanzu CLI. I have been trying to examine what is happening using "strace" to see if I can determine what sort of request has been sent to AD, but this is also complex. I guess something like wireshark or other network sniffer could help.

There is also very little documentation around what to place in these fields. Some further guidance would be helpful.

Environment:

TKG CLI version (from “tkg version”): v1.4

Target Infrastructure and version (e.g. vsphere:0.6.0, aws:0.5.0): vSphere 7.0U2

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [x] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [x] User Experience

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

**Comment by miclettej** [comment]
Unable to reproduce

**Comment by miclettej** [comment]
hi @sabbey37 reassigning to you for triage. Please reach out to any CLI team member or reassign if needed. Thank you!

**Comment by sabbey37** [comment]
Hey, @raymondz1 and @miclettej .  I think there may have been some miscommunication with this issue.  I saw in the MR that the text was added to the existing tooltip for `Username` for the `User Search Attributes` and `Name Attribute` for the `Group Search Attributes`.  This is misleading.  Pinniped/TKG IAM utilize whatever the user provides in this configuration, that is not the issue.  The search is hardcoded on **the Kickstart UI side** to utilize the filter + CN for user search (https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/tkg/ldap/client.go#L111) and the filter + OU for Group Search (https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/tkg/ldap/client.go#L180).

The issue lies in the LDAP verification performed by the Kickstart UI, which **does not utilize Pinniped/TKG IAM tooling at all**, it just performs a normal HTTP call.  I was imagining the tooltip text would be added to this page:
<img width="513" alt="Screen Shot 2022-01-28 at 12 56 02 PM" src="https://user-images.githubusercontent.com/41928668/152551900-8c9318cb-806c-4f2a-b2a8-e21f2af7f572.png">
For `TEST USER NAME` the tooltip should indicate this is the `CN` attribute for the user, and for `TEST GROUP NAME`, the tooltip should indicate that this is the `OU` attribute for the group.
